retrieval_service/sql-proxy/cloudsql.tests.cloudbuild.yaml

```diff
@@ -0,0 +1,161 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  - id: "docker-build"
+    name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-t", "${_IMAGE_NAME}", "retrieval_service/sql-proxy/."]
+
+  - id: "docker-push"
+    name: "gcr.io/cloud-builders/docker"
+    args: ["push", "${_IMAGE_NAME}"]
+
+  - id: "docker-layer"
+    name: "gcr.io/cloud-builders/docker"
+    entrypoint: /bin/bash
+    args:
+      - '-c'
+      - |
+        echo "FROM $_IMAGE_NAME
+        COPY --from=gcr.io/cloud-sql-connectors/cloud-sql-proxy /cloud-sql-proxy /cloudsql/cloud-sql-proxy" > Dockerfile-proxy;
+
+        docker build -f Dockerfile-proxy -t ${_IMAGE_NAME}-proxy .
+
+  - id: Install dependencies
+    name: python:3.11
+    dir: retrieval_service
+    entrypoint: pip
+    args:
+      [
+        "install",
+        "-r",
+        "requirements.txt",
+        "-r",
+        "requirements-test.txt",
+        "--user",
+      ]
+
+  - id: Create database
+    name: ${_IMAGE_NAME}-proxy
+    dir: retrieval_service/sql-proxy
+    entrypoint: /bin/bash
+    secretEnv:
+      - CLOUDSQLUSER # Use built-in env vars for database connection
+      - CLOUDSQLPASSWORD
+    args:
+      - "-c"
+      - |
+        /cloudsql/cloud-sql-proxy --port ${_DATABASE_PORT} ${_INSTANCE_CONNECTION_NAME} & sleep 2;
+        psql -h ${_DATABASE_HOST} -c "CREATE DATABASE ${_DATABASE_NAME};" || echo "Database '${_DATABASE_NAME}' already exists."
+        psql -h ${_DATABASE_HOST} -d ${_DATABASE_NAME} -c 'CREATE EXTENSION vector;' || echo "Extension 'vector' already exists."
+
+  - id: Initialize data
+    name: python:3.11
+    dir: retrieval_service
+    secretEnv:
+      - CLOUDSQLUSER
+      - CLOUDSQLPASSWORD
+    entrypoint: /bin/bash
+    args:
+      - "-c"
+      - |
+        # Create config
+        cp example-config.yml config.yml
+        sed -i "/127.0.0.1/d" config.yml
+        sed -i "s/postgres/cloudsql-postgres/g" config.yml
+        sed -i "s/my_database/${_DATABASE_NAME}/g" config.yml
+        sed -i "s/my-user/$$CLOUDSQLUSER/g" config.yml
+        sed -i "s/my-password/$$CLOUDSQLPASSWORD/g" config.yml
+        echo -e '\n  project: "'$PROJECT_ID'"' >> config.yml
+        echo '  region: "'${_CLOUDSQL_REGION}'"' >> config.yml
+        echo '  instance: "'${_CLOUDSQL_INSTANCE}'"' >> config.yml
```

Comment:
Do we want to add another example config file in order make this easier? We could have rename the current to postgres-config.yml and add another maybe named cloudsql-config.yml